torch/csrc/distributed/rpc/tensorpipe_agent.h

```diff
@@ -455,6 +455,10 @@ class TORCH_API TensorPipeAgent : public RpcAgent {
   // e.g. updates to (workerIdToInfo_, workerNameToInfo_, workerNameToURL_)
   mutable std::mutex groupMembershipMutex_;
 
+  // CV to watch for changes after groupMembership is updated
+  mutable std::mutex unknownWorkerMutex_;
```

Comment:
I don't see what this new mutex is protecting. It's fine to share the same mutex across CVs.
We must always document what fields, conditions and state transitions are protected by a mutex.
Why do we need a new mutex and where does it fit in the locking hierarchy.